openselery/openselery.py

```diff
@@ -0,0 +1,291 @@
+        "coinbase_token": 'COINBASE_TOKEN',
+        "coinbase_secret": 'COINBASE_SECRET',
+    }
+    __default_config_template__ = {
+        "bitcoin_wallet": "",
+
+        "dryrun": True,
+        "include_dependencies": False,
+        "include_self": True,
+        "include_tooling_and_runtime": False,
+        "min_contributions": 1,
+        "check_equal_privat_and_public_wallet": True,
+        "skip_email": True,
+        "email_note": "Fresh OpenCelery Donation",
+        "btc_per_transaction": 0.000002,
+        "contributor_payout_count": 1,
+        "total_payout_per_run": 0.000002,
+    }
+    __secure_config_entries__ = ["libraries_api_key", "github_token", "coinbase_token", "coinbase_secret",
+                                 "coinbase_secret"]
+
+    def __init__(self, d={}):
+        super(OpenSeleryConfig, self).__init__()
+        self.apply(self.__default_config_template__)
+        self.apply(d)
+
+    def apply(self, d):
+        self.__dict__.update(d)
+
+    def applyYaml(self, path):
+        yamlDict = yaml.safe_load(open(path))
+        ### ensure type of loaded config
+        for k, v in yamlDict.items():
+            t1, v1, t2, v2 = type(v), v, type(getattr(self, k)), getattr(self, k)
+            if t1 != t2:
+                raise ValueError("Configuration parameter '%s' has failed type check! 's'<'%s'> should be 's'<'%s'>" % (
+                    k, v1, t1, v2, t2))
+        ### special evaluations 
+        if not self.total_payout_per_run / self.contributor_payout_count == self.btc_per_transaction:
+            raise ValueError("Payout values do not match")
+        self.apply(yamlDict)
+
+    def applyEnv(self):
+        try:
+            environmentDict = {k: os.environ[v] for k, v in self.__default_env_template__.items()}
+            self.apply(environmentDict)
+        except KeyError as e:
+            raise KeyError("Please provide environment variable %s" % e)
+
+    def __repr__(self):
+        ### make config safe for printing
+        # secureEntries = {k: "X"*len(os.environ[v]) for k, v in self.__default_env_template__.items()}
+        secureEntries = {k: "X" * len(getattr(self, k)) for k in self.__secure_config_entries__}
+        secureDict = dict(self.__dict__)
+        secureDict.update(secureEntries)
+        return str(" --\n%s\n --" % secureDict)
+
+
+class OpenSelery(object):
+    def __init__(self, silent=False):
+        super(OpenSelery, self).__init__()
+        ### set our openselery project dir, which is '../../this_script' 
+        self.seleryDir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        self.silent = silent
+        self.librariesIoConnector = None
+        self.githubConnector = None
+        ### start initialization of configs
+        self.initialize()
+
+    def __del__(self):
+        self.logNotify("Feel free to visit us @ https://github.com/protontypes/openselery")
+        self.logNotify("Done")
+        
+    def initialize(self):
```

Comment:
only a personal preference, because it is now possible to re-initialize an allready constructed object via a known method without deleting (destructing) it. 

so calling object.initialize() is somethign anyone should be able to do to re-read configurations in case they have changed at runtime without restarting a program or removing known constructs

not really necessary